### PR TITLE
Disable brace mismatch hightlight in comment

### LINKED ISSUE #84556: Brace mismatch highlight appears for comments as well
### Godot version

4.2.beta1

### System information

Windows 11 - Godot 4.2.beta1

### Issue description

image
image

for some reason, the bracket is only highlighted if the caret is next to the bracket

### Steps to reproduce

1 - Make a comment
2 - End it with ) and dont include an opening bracket (
3 - Put your caret next to it

### Minimal reproduction project

N/A: too trivial, just open any project and make a comment

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes #84556

But the fix is very brute force and hard coded, let's see if there is more elegant ways.

### CODE REVIEW

**Comment by Chubercik** [comment]
This doesn't seem to fix the issue, only one edge case:

image

If the line doesn't start with `#` (so basically all inline comments), the issue at hand is still present:

image

**Review by Chubercik — CHANGES_REQUESTED** [review]
To test if the fix works, one should consider the following checks:

1. Comment at the beginning of the line (here - passes):
```gdscript
# test)
```

2. Above example, but with an indent:
```gdscript
    # test)
```

3. An inline comment:
```gdscript
var test: int = 0  # test)
```

File: scene/gui/text_edit.cpp [hunk 1]
```diff
@@ -1243,6 +1243,12 @@ void TextEdit::_notification(int p_what) {
 						if (char_pos >= xmargin_beg) {
 							if (highlight_matching_braces_enabled) {
 								for (int c = 0; c < carets.size(); c++) {
+									// Brace mismatch highlighting shouldn't happen within comments.
+									String lineData = get_line(line);
+									if (lineData.begins_with("//") || lineData.begins_with("#")) {
```
[the remark this answers is not in the document]
Comment (AThousandShips) [reply scene/gui/text_edit.cpp:1248]: This isn't just for GDScript

[the remark this answers is not in the document]
Comment (Chubercik) [reply scene/gui/text_edit.cpp:1248]: I see, fair point

**Comment by jsjtxietian** [comment]
Thanks for all the suggestion!

**Comment by Paulb23** [comment]
Please take a look at my comment in #53364 the fix is quite a lot more involved then it first seems.

**Comment by jsjtxietian** [comment]
Yes, the brace highlight should not exist in `TextEdit` and it needs to be moved to `CodeEdit`.
